### PR TITLE
Allow building Merlin with older versions of `docker`.

### PR DESCRIPTION
**Pull request by bashimao**
Modified images according to a trick discovered by @EmmaQiaoCh  , which allows compiling the images on ARM machines with old versions of `docker` and versions that do not come with `buildx`.

### CODE REVIEW

File: docker/dockerfile.merlin [hunk 1]
```diff
@@ -225,7 +226,10 @@ ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${JAVA_HOME}/lib:${JAVA_HOME}/lib/server
 # Binaries
 COPY --chown=1000:1000 --from=build /usr/local/bin/cmake /usr/local/bin/
 COPY --chown=1000:1000 --from=build /usr/local/bin/pytest /usr/local/bin/
-COPY --chown=1000:1000 --from=build /usr/local/bin/perf_* /usr/local/bin/
+# NOTE 2023-09: fil-perf_analyzer is not available on ARM. Some docker versions flag an error if
+#               there is not a single source file to copy. To avoid this, we als specify a small
+#               dummy file.
+COPY --chown=1000:1000 --from=build /usr/local/bin/cudaCheck /usr/local/bin/perf_* /usr/local/bin/
```
[the remark this answers is not in the document]
Comment (bashimao) [reply docker/dockerfile.merlin]: Thanks for your comment. I independently found that problem today as well.  We previously fixed that, but the Merlin team reverted it. I believe relying on the `pip` package is flawed anyway. Therefore, I just pushed a fix that resolves the problem in a "hopefully" more proper manner. Can you review again?

**Comment by bashimao** [comment]
What this actually fixes:
1. It fixes the `perf_analyzer` not being there on ARM64 systems.
2. If suppresses the error where the Triton `fil` backend doesn't exist in the base image with older versions of `docker`.
3. It fixes a bunch of typos.
4. I removed the check that prevented installing `python3-libnvinfer` on ARM64 systems. That was a legacy issue that got resolved through fixes in the upstream image. Since the `tritonserver` switched to Ubuntu 22.04 as base image, `libnvinfer` is available.

**Review by EmmaQiaoCh — APPROVED** [review]

**Comment by EmmaQiaoCh** [comment]
The copied PR is merged

**Pull request closed without merging**